nix/module.nix

```diff
@@ -0,0 +1,89 @@
+inputs: { config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.asus-numberpad-driver;
+  defaultPackage = inputs.self.packages.${pkgs.system}.default;
+
+  # Function to convert configuration options to string
+  toConfigFile = cfg: builtins.concatStringsSep "\n" (
+    [ "[main]" ] ++ lib.attrsets.mapAttrsToList (key: value: "${key} = ${value}") cfg.config
+  );
+
+  # Writable directory for the config file
+  configDir = "/etc/asus-numberpad-driver";
+in {
+  options.services.asus-numberpad-driver = {
+    enable = lib.mkEnableOption "Enable the Asus Numberpad Driver service.";
+
+    layout = lib.mkOption {
+      type = lib.types.str;
+      default = "up5401ea";
+      description = "The layout identifier for the numberpad driver (e.g. up5401ea). This value is required.";
+    };
+
+    config = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      default = {};
+      description = ''
+        Configuration options for the numberpad driver.
+        These options will be written to a configuration file for the driver.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ defaultPackage ];
+
+    # Ensure the writable directory exists
+    systemd.tmpfiles.rules = [
+      "d ${configDir} 0755 root root -"
+    ];
+
+    # Write the configuration file to the writable directory
+    environment.etc."asus-numberpad-driver/numberpad_dev".text = toConfigFile cfg;
+
+    # Enable i2c
+    hardware.i2c.enable = true;
+
+    # Add groups for numpad
+    users.groups = {
+      uinput = { };
+      input = { };
+      i2c = { };
+    };
+
```

Comment:
@scientiac About features would be great to have at least an associated `CALC` key to the usually used calculator on NixOs or just some calculator app so others will know the way how to set up this. Other features rely on `udev`. Any way how to implement them? (I see them like minor feature, opening calc app as necessary to implement)

Reply:
I have been using this in gnome (wayland) and the calculator app opens with the gesture (currently doesn't toggle but just opens on second gesture as well).

Reply:
I see as necessary to implement toggling (equivalent of `install_calc_toggle.sh` using gsettings) in still installation.

Reply:
I need to look into it. I quite dont know how to do that yet.

Reply:
What is the default behavior of the driver? Doing the swipe gesture to close the calculator app closes the backlight or not? I managed to make it work, it toggles the calculator app but doesnt toggle the numpad (Is this expected?).

Reply:
@scientiac Yes, please do not toggle numpad. Only calc app. What Is default driver behaviour on Windows I do not know but eventualy the change would be done in my code and is up to me.

Reply:
I have found a way to make it work declaratively. 

As it seems to only require the `XF86Calculator` to have a shortcut to a toggle script for gnome-calculator app.

But it isn't tied to the service by default but is entirely upto the user to enable. I guess putting a way to set it up in the NixOS portion of readme should be fine for NixOS users.